text/trace/0226-sampling-random-traceids.md

```diff
@@ -0,0 +1,185 @@
+by padding with 0s.  For example, the t-value string "003f"
+corresponds with a the 14-hex-digit string "003f0000000000".
+
+Head samplers and tail samplers alike can be implemented simply by
+tersting whether the least-significant 7 bytes of the TraceID are
+lexicgraphically less-than-or-equal to the sampling threshold.  Note
+that this comparison may be carried out directly on hex digits or on
+binary data using simple string or bytes comparison.
+
+Modifying an in-SDK Sampler to perform this calculation is a simple
+change relative to setting p-value for sampled spans.  For tail
+samplers, a span processor can simply pass through all spans where the
+least-significant 7-bytes of TraceID are less-than-or-equal to the
+configured threshold.  When the span passes, it has its TraceState
+t-value set to the configured threshold for use in Span-to-Metrics
+pipelines.
+
+### Converting between Thresholds and Probabilities
+
+Sampling probabilities in the range (0, 1] can be mapped onto 56-bit
+encoded t-values in the range [0, 0xffffffffffffff].  For a given
+sampling threshold, the corresponding probability is expressed as a
+fraction `(T+1)/2^56` (i.e., sampling threshold plus one divided by
+2^56).
+
+Note that IEEE double-width floating point numbers use 52 bits of
+significand, so not all sampling thresholds have corresponding
+floating point values that the user might be able to express.
+
+For SDKs and Span processors to implement consistent probability
+sampling, OpenTelemetry should define how to compute a sampling
+threshold from a floating point number and in the reverse direction,
+how to compute a floating point number from a threshold.  Combined,
+these rules allow simple sampling logic to be easily translated into
+probabilities or adjusted counts for use in a Span-to-Metrics
+pipeline.
+
+#### Probability to Hex Threshold
+
+Note that the procedure here only works for probabilities greater than
+or equal to 2^-52.
+
+To convert from a floating point number to the nearest threshold as a
+14-byte hex string:
+
+```
+func ProbabilityToThreshold(prob float64) string {
+	return fmt.Sprintf("%.14x", math.Nextafter(prob+1, 0))[4:18]
+}
+```
+
+Note that this can be truncated after one or more non-zero digits,
+leaving a more-compact encoding of a sampling probability that is
+nearby.
+
+Note that the threshold is rounded down, it will be slightly smaller
+than the configured probabilty in cases where the probability cannot
+be exactly represented in 56 bits.
+
+#### Hex Threshold to Probability
+
+To convert a hex threshold string to the corresponding probability, we
+perform that opposite of the above.
+
+```
+func ThresholdToProbability(thresh string) float64 {
+    parsed, _ := strconv.ParseFloat("0x1."+thresh[:13]+"p+00", 64)
+	return math.Nextafter(parsed, 2) - 1
+}
+```
+
+Note that these transformations are not always reversible, since
+floating point numbers have less precision.  Note that only 13 bytes
```

Comment:
double-precision has effectively 53bits of precision (the first 1bit is implicitly stored + 52bits mantissa), so the transformation could potentially use 13bytes + 1bit